component/src/action_handler/actions/undelegate.rs

```diff
@@ -78,6 +86,11 @@ impl ActionHandler for Undelegate {
     async fn execute(&self, state: &mut StateTransaction) -> Result<()> {
         tracing::debug!(?self, "queuing undelegation for next epoch");
         state.stub_push_undelegation(self.clone());
+        // Register the undelegation's denom, so we clients can look it up later.
+        state
+            .register_denom(&self.unbonding_token().denom())
+            .await?;
+        // TODO: should we be tracking changes to token supply here or in end_epoch?
```

Comment:
As a followup to this, we should nail down how we track changes to the token supply, and if we can do so -- the undelegate action reveals the amount, for now, but the undelegateclaim doesn't, so we may not be able to precisely track the supply of all the tokens precisely.